dulwich/graph.py

```diff
@@ -20,11 +20,73 @@
 
 """Implementation of merge-base following the approach of git."""
 
-from collections import deque
-from typing import Deque
+from collections.abc import MutableMapping
+from collections import OrderedDict
 
+from heapq import heappush, heappop
 
-def _find_lcas(lookup_parents, c1, c2s):
+# Because opening an sha file from the object_store, reading it,
+# then decoding its will be a slow operation, we want to create
+# a cache of Commit objects so that repeat parent and commit time
+# lookup (which are inherent when looking for a common ancestor)
+# will be much faster, so we need to implement a simple Commit cache
+# using an OrderedDict
+
+
+class CommitCache(MutableMapping):
+
+    def __init__(self, maxsize):
+        self.maxsize = maxsize
+        self.d = OrderedDict()
+
+    def __getitem__(self, key):
+        self.d.move_to_end(key)
+        return self.d[key]
+
+    def __setitem__(self, key, value):
+        if key in self.d:
+            self.d.move_to_end(key)
+        elif len(self.d) == self.maxsize:
+            self.d.popitem(last=False)
+        self.d[key] = value
+
+    def __delitem__(self, key):
+        del self.d[key]
+
+    def __iter__(self):
+        return self.d__iter__()
+
+    def __len__(self):
+        return len(self.d)
+
+    def __contains__(self, key):
+        return key in self.d
+
+
+# priority queue using builtin python minheap tools
+# why they do not have a builtin maxheap is simply ridiculous but
+# liveable with integer time stamps using negation
+class WrkLst(object):
```

Comment:
Please just spellout the name? Is this WorkList?

Reply:
Yes I will make that change.